src/components/ModelingMachineProvider.tsx

```diff
@@ -1497,6 +1497,46 @@ export const ModelingMachineProvider = ({
             return result
           }
         ),
+        'set-up-draft-arc-three-point': fromPromise(
+          async ({ input: { sketchDetails, data } }) => {
+            if (!sketchDetails || !data)
+              return reject('No sketch details or data')
+            sceneEntitiesManager.tearDownSketch({ removeAxis: false })
+            const result = await sceneEntitiesManager.setupDraftArcThreePoint(
+              sketchDetails.sketchEntryNodePath,
+              sketchDetails.sketchNodePaths,
+              sketchDetails.planeNodePath,
+              sketchDetails.zAxis,
+              sketchDetails.yAxis,
+              sketchDetails.origin,
+              data
+            )
+            if (err(result)) return reject(result)
+            await codeManager.updateEditorWithAstAndWriteToFile(kclManager.ast)
```

Comment:
Small warning about this maybe shouldnt be awaited?

Reply:
Oh right, so what do we do add `// eslint-disable-next-line @typescript-eslint/no-floating-promises` above, or do a `.catch(reportRejection)` ?

Reply:
I did the former, but can change it again 11be6329c0ea753b5807f42650f7457abe1a27c5.